README.md

```diff
@@ -2,31 +2,109 @@
 
 ## Learning Goals
 
-- Learning Goal 1
-- Learning Goal 2
+- Define Java Servlet
+- Define Java Servlet Container
+- Define Servlet Life Cycle 
 
 ## Introduction
 
-A 1-2 sentence summary of what will be covered.
+A Java **Servlet** is a class that is often used for developing dynamic web applications.
+Typically, a servlet takes an HTTP request from a browser or other HTTP client, generates dynamic content,
+and provides an HTTP response back to the browser/client.
+For example, we can implement a Servlet to receive input
+from a user through an HTML form, query records from a database based on the user input,
+and generate a dynamic web page as a response.
 
-## Topic 1
+## Java Servlets
 
-Discuss the topic of the reading. Break into sections and sub-sections as
-appropriate to help with the reader's orientation with the material and future
-referencing.
+A Java servlet class must implement the `javax.servlet.Servlet` interface. 
+This interface specifies methods to initialize a servlet, process requests, get information of a servlet,
+and terminate a servlet instance.
 
-## Topic 2
+The `Servlet` interface includes the following methods:
 
-Discuss the topic of the reading. Break into sections and sub-sections as
-appropriate to help with the reader's orientation with the material and future
-referencing.
+- init(...): Initialize the servlet.  
+- destroy(...): Terminate the servlet.  
+- service(...): Receive HTTP requests and, by default, dispatch them to the appropriate doXXX() methods.  
+- getServletConfig() - Returns an object that contains initialization and startup parameters.  
+- getServletInfo(...): Retrieve information about the servlet.  
+
+For Web applications, we implement the Servlet interface by extending the `javax.servlet.http.HttpServlet` abstract class. 
+For protocol-independent servlets, we extend the `javax.servlet.GenericServlet` class.
+
+The `HttpServlet` class includes the following methods (among others not shown):
+
+- doGet(...): Execute an HTTP GET request.  
+- doPost(...): Execute an HTTP POST request.  
+- doPut(...): Execute an HTTP PUT request.  
+- doDelete(...): Execute an HTTP DELETE request.  
+
+A servlet class that extends `HttpServlet` would override some or all of these methods
+based on the application requirements. For example, we might override the `doGet()` method
+to accept user input from an HTML form and generate a response based on the form data.
+
+The `service` and various `do...()` methods take as parameters an `HttpServletRequest` instance and an `HttpServletResponse` instance.
```

Comment:
It could just be me but this sentence seems a little awkward. Maybe

"The `service` and various `do...()` methods take the following as parameters:" (and then list the bullet points)

Reply:
Agreed! Updated to just list the two parameters.